### ISSUE TITLE
IntroduceVariable should not trigger on the name in a local declaration

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#39536]
Repro:
```C#
class C
{
    int M()
    {
        int x[||] = M();
        System.Console.Write(x);
    }
}
```

An "introduce local" refactoring is offered.
I expect there would be no "introduce local" offered, and "inline temporary variable" would be the top choice.

![image](https://user-images.githubusercontent.com/12466233/67611454-a50b4600-f74e-11e9-98c3-c85e22eb3607.png)

FYI @jasonmalinowski @petrroll

**Comment by CyrusNajmabadi** [comment]
Notes: 
1. Inline-temp should be the top value.  Def agree with that.
2. The change to add another 'introduce local' here was intentional and based on people stating that they wanted this facility to clone a declared variable.  I personally don't see a ton of value here, but permanently it's something people expect after using resharper.

**Comment by petrroll** [comment]
Not sure if both of these are refactorings but these would be correctly ordered (as per note 1) it this event went through: https://github.com/dotnet/roslyn/pull/38271

**Comment by petrroll** [comment]
Ideally "inline temporary" would be on top because the cursor is closer to variable declaration than it's expression. Unfortunately ordering based on caret location & applicableToSpan for `refactorings` hasn't been merged in due to issues outlined in #38271.

Since both of these refactorings are medium priority (I assume) a quick fix would be to change their LoadBeforeAttribute (or whatever its name is) to impose static (context oblivious) ordering. I'd, however, vote strongly for trying to finish work on #38271 as it will solve loads of problems that might come down the road.

**Comment by CyrusNajmabadi** [comment]
Closing out.  This was intentional and done based on user requests.
